src/Symfony/Component/Form/Form.php

```diff
@@ -795,6 +795,25 @@ public function getErrors($deep = false, $flatten = true)
         return new FormErrorIterator($this, $errors);
     }
 
+    /**
+     * {@inheritdoc}
+     */
+    public function clearErrors(bool $deep = false)
```

Comment:
missing return type hint (same in the interface)

Reply:
actually I am not even sure if it makes sense that this method returns anything

Reply:
All form methods are chainable, so it makes sense to keep it here.

Reply:
@xabbuh see https://github.com/symfony/symfony/pull/27580#discussion_r195900434

Reply:
I think it's time to decide this once and for all in https://symfony.com/doc/current/contributing/code/standards.html

Reply:
Well, fluent methods are a pain for return types. so even if we want to have as much return types as possible for new APIs, I'm not sure we want them for fluent APIs.

Reply:
@xabbuh @stof I have removed `return $this;` from the method for a few reasons:

1. You're right, this method doesn't really need to be chainable.
2. [ClearableTokenStorageInterface](https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Security/Csrf/TokenStorage/ClearableTokenStorageInterface.php) (a new and similar interface introduced in SF4) does not return anything, so perhaps we should be consistent here.
3. It seems like there's no definitive rule on how to type hint interface methods that return `$this` / `self`. I don't want to set a bad precedent here.

Reply:
Then you should add `: void` in interface and class

Reply:
But I am for consistency with `FormInterface` which has `@return $this` on methods like `addError`. Although I agree that fluent interfaces are pain.

Reply:
**I've changed my mind and added `return $this;` back in.**  I could see developers wanting to call `$form->clearErrors()->createView()` so this definitely has value.

The last unanswered question is what to do about return types.  Our ultimate goal is to have `Form::clearErrors()` return that original `Form` instance.  Here's what the [Return Type Declarations RFC](https://wiki.php.net/rfc/return_types#variance_and_signature_validation) has to say about our current situation:

> The enforcement of the declared return type during inheritance is invariant; this means that when a sub-type overrides a parent method then the return type of the child must exactly match the parent and may not be omitted.

This is why using `: self` on the interface requires the `Form` method to use `: ClearableErrorsInterface` which is not ideal.

However, the RFC goes on to say:

> If the parent does not declare a return type then the child is allowed to declare one.

Because this is a valid approach, I think **omitting the return type from the interface but using `: self` on the `Form`** makes the most sense in this situation:

1. A return type on the interface is pointless because there are literally no other methods on `ClearableErrorsInterface`.  Is somebody really going to call `$form->clearErrors()->clearErrors()->clearErrors()`?
2. This gives us the functionality we want (accurate return type hint on `Form`) despite PHP's current limitations with invariant return types.